queries/__init__.py

```diff
@@ -253,6 +253,546 @@ def convert(m):
+            text.insert(-2, "og")
+
+    # Fix e.g. "milljónir milljarðar" -> "milljónir milljarða"
+    number_string: str = re.sub(
+        r"(\S*(jónir|jarð[au]r?)) (\S*(jarð|jón))[ia]r", r"\1 \3a", " ".join(text)
+    )
+
+    return number_string
+
+
+def number_to_text(n: int, gender="hk") -> str:
+    """
+    Convert an integer into written Icelandic text in given gender (hk, kk, kvk).
+    Example:
+        302 -> "þrjú hundruð og tvær" (gender="kvk")
+        501 -> "fimm hundruð og einn" (gender="kk")
+    """
+    num_str = number_to_neutral(n)
+
+    if gender =="hk":
+        return num_str
+
+    nums = num_str.split()
+
+    if gender=="kk" and nums[-1] in _NUM_NEUT_TO_KK:
+        nums[-1] = _NUM_NEUT_TO_KK[nums[-1]]
+
+    elif gender=="kvk" and nums[-1] in _NUM_NEUT_TO_KVK:
+        nums[-1] = _NUM_NEUT_TO_KVK[nums[-1]]
+
+    return " ".join(nums)
+
+
+def float_to_text(f: float = 0.0, gender="hk") -> str:
+    """
+    Convert a float into written Icelandic text in given gender (hk, kk, kvk).
+    Example:
+        -0.02 -> "mínus núll komma núll tveir" (gender="kk")
+    """
+    out_str: str = ""
+    # To prevent edge cases like -0.2 being translated to
+    # "núll komma tvö" instead of "mínus núll komma tvö"
+    if f < 0:
+        out_str = "mínus "
+
+    first, second = str(f).split(".")
+
+    # Number before decimal point
+    out_str += number_to_text(abs(int(first)), gender)
+    out_str += " komma "
+
+    # Numbers after decimal point
+    for digit in second:
+        if digit == "0":
+            out_str += "núll "
+        else:
+            digit_str = _SUB_20_NEUTRAL.get(int(digit), "")
+            if gender == "kk" and digit_str in _NUM_NEUT_TO_KK:
+                out_str+= _NUM_NEUT_TO_KK[digit_str]
+            elif gender == "kvk" and digit_str in _NUM_NEUT_TO_KVK:
+                out_str+= _NUM_NEUT_TO_KVK[digit_str]
+            else:
+                out_str += digit_str
+            out_str += " "
+
+    return out_str.strip()
+
+
+def year_to_text(year: int, after_christ: bool = False) -> str:
+    """
+    Write year as text in Icelandic.
+    Negative years automatically append "fyrir Krist" to the text.
+    If after_christ is True, add "eftir Krist" after the year.
+    """
+    suffix: str = ""
+    text: List[str] = []
+
+    if year < 0:
+        suffix = " fyrir Krist"
+        year = -year
+
+    elif year > 0 and after_christ:
+        suffix = " eftir Krist"
+
+    # People say e.g. "nítján hundruð þrjátíu og tvö"
+    # instead of "eitt þúsund níu hundruð þrjátíu og tvö"
+    # for years between 1100-2000
+    if 1100 <= year < 2000:
+        hundreds, digits = divmod(year, 100)
+
+        text.append(_SUB_20_NEUTRAL[hundreds])
+        text.append("hundruð")
+        text.append(number_to_neutral(digits))
+
+    # Other years are spoken like regular numbers
+    else:
+        text.append(number_to_neutral(year))
+
+    return " ".join(text) + suffix
+
+
+FYRSTUR_STRONG_DECL = {
```

Comment:
Should have leading underscore for consistency?

A Mapping type declaration would be good here, to enforce constness

Reply:
Fixed